packages/shylu/shylu_dd/frosch/src/Tools/FROSch_SubdomainSolver_def.hpp

```diff
@@ -215,7 +215,97 @@ namespace FROSch {
+  FROSCH_ASSERT(RepeatedMaps.size()==dofOrderings.size(),"RepeatedMaps.size()!=dofOrderings.size()");
+  TLBP = Teuchos::rcp(new TwoLevelBlockPreconditioner<SC,LO,GO,NO>(K_,ParameterList_));
+  TLBP->initialize(ParameterList_->get("Dimension",3),
+                   dofsPerNodeVector,dofOrderings,
+                   ParameterList_->get("Overlap",1),
+                   RepeatedMaps,
+                   nullSpaceBasisVec);
+
+} else if(!ParameterList_->get("SolverType","Amesos2").compare("TwoLevelPreconditioner")) {
+
+  Teuchos::RCP< const Teuchos::Comm< int > > TC = K_->getMap()->getComm();
+  Teuchos::ArrayRCP<Teuchos::RCP<const Xpetra::Map<LO,GO,NO> > > RepeatedMaps(1);
+  Teuchos::ArrayRCP<Teuchos::RCP<const Xpetra::Map<LO,GO,NO> > > NodesMaps(1);
+
+  UNVecPtr dofsPerNodeVector(1);
+  Teuchos::ArrayRCP<DofOrdering> dofOrderings(1);
+  Teuchos::ArrayRCP<Teuchos::ArrayRCP<Teuchos::RCP<const Xpetra::Map<LO,GO,NO> > > > dofsMapsVec = Teuchos::null;
+  Teuchos::ArrayRCP<Teuchos::RCP<Xpetra::Map<LO,GO,NO> > > MainCoarseMapVector = Teuchos::null;
+
+
+  //FROSCH_ASSERT(ParameterList_->isParameter("Repeated Map Vector"),"Currently TwoLevelBlockPreconditioner cannot be constructed without Repeated Maps Vector ");
+
+  if(ParameterList_->isParameter("Repeated Map Vector")) {
+    RepeatedMaps = ExtractVectorFromParameterList<Teuchos::RCP<const Xpetra::Map<LO,GO,NO> > >(*ParameterList_,"Repeated Map Vector");
+  }
+
+  if(ParameterList_->isParameter("Nodes Map Vector")) {
+    NodesMaps = ExtractVectorFromParameterList<Teuchos::RCP<const Xpetra::Map<LO,GO,NO> > >(*ParameterList_,"Nodes Map Vector");
+  }
+
+  if(ParameterList_->isParameter("DofsPerNode Vector")) {
+    dofsPerNodeVector = ExtractVectorFromParameterList<UN>(*ParameterList_,"DofsPerNode Vector");
+  }
+
+  if(ParameterList_->isParameter("DofOrdering Vector")) {
+    dofOrderings = ExtractVectorFromParameterList<DofOrdering>(*ParameterList_,"DofOrdering Vector");
+  }
+
+  if(ParameterList_->isParameter("Main Map Vector")) {
+      MainCoarseMapVector = ExtractVectorFromParameterList<Teuchos::RCP<Xpetra::Map<LO,GO,NO> > >(*ParameterList_,"Main Map Vector");
+  }
+  if(ParameterList_->isParameter("Dofs Maps Vector")) {
+      dofsMapsVec = ExtractVectorFromParameterList<Teuchos::ArrayRCP<Teuchos::RCP<const Xpetra::Map<LO,GO,NO> > >>(*ParameterList_,"Dofs Maps Vector");
+  }
+
+
+  ConstXMultiVectorPtr nodeList = null;
+  GOVecPtr dirichletBoundaryDofs = null;
+  ConstXMultiVectorPtr nullSpaceBasisVec = null;
+
+  TLP = Teuchos::rcp(new TwoLevelPreconditioner<SC,LO,GO,NO>(K_,ParameterList_));
+  TLP->initialize(ParameterList_->get("Dimension",3),
+                  dofsPerNodeVector[0],
+                  ParameterList_->get("Overlap",1),
+                  nullSpaceBasisVec,
+                  nodeList,
+                  dofOrderings[0],
+                  RepeatedMaps[0],
+                  dofsMapsVec[0],
+                  dirichletBoundaryDofs);
+    }else {
```

Comment:
Does the three-level implementation also work for the block variant?

Reply:
Not yet, but I wanted it to be prepared